### ISSUE TITLE
Question about power and capacitor wiring

### ISSUE THREAD

**Issue by gingters** [hzeller/rpi-rgb-led-matrix#997]
Hi. Since my last question I now got myself two 16x32 P10 Outdoor panels, and only then I read the wiring and "A word about power" section 🙈.

So, to be sure everything goes right I now ordered a new power source (5V 10A, 50W) as my current supply is not strong enough, some capacitors (3300µF/25V) and the suggested active connector board from china (2 boards for 3 USD each, plus 4,32 shipping, wasn't that bad after all).

Just to be sure, as my electronics knowledge isn't that good (as in, I never learned a lot, and when I did that was over 2 decades ago ;) ): You (@hzeller) suggested two of these capacitors in parallel, directly at the power connector on the panels.

My panel has one power connector with four pins, 2 VCC and two GND. I double-checked that both VCC pins are connected together and both GND pins are too, so I think we only have two of each to evenly distribute the power between both pins.

I would now solder one capacitor between the inner VCC and GND and the other capacitor between the outer VCC and GND (to make sure they don't get shorted out by crossing their connectors).

So that would look like

```
  ------ CAP1 ------
  |+               |-
  |      CAP2      | 
  |     |+   |-    |
VCC   VCC   GND   GND
```

Is that correct?

**Comment by hzeller** [comment]
Yes, this looks good.
If you only have one or two panels connected, maybe your power supply takes care enough of the power surges so you might not even need the capacitors. The longer the cables and the more panels are connected, the more likely you need capacitors.

**Comment by gingters** [comment]
Thanks for the fast answer!

This is great. Cables won't be long from power supply to panel, and I really hope to only need one panel per display, two at max. And also I don't plan on having them show full white a lot ;)

They should only show a few letters and numbers in red, green or yellow, sometimes the time and if I really need two for the width, then I maybe can display a bit of a scrolling text (that would probably be not very convenient /readable on only one panel).

**Comment by gingters** [comment]
Before I open up a new issue, I'd like to reopen here and ask another question that - I think - fits well here:

Is there a way to determine upfront if my panel needs 5V signal input or if the 3.3V from the Raspi GPIO is enough?

That is: Should I wait until the active driver boards arrive or is there a good chance that I can start coding and testing without the board right away?

Also:
I now ordered a tad more beefy power supply for my two panels (https://www.amazon.de/gp/product/B07DQKM9P7/ref=ppx_yo_dt_b_asin_title_o03_s00?ie=UTF8&psc=1). The cables that came with the panel are 60cm from power supply to first panel and then from that connector they extend another 40cm to the second panel. Is that short enough to avoid the capacitors for the time being? (I'm a bit reluctant, as my soldering skills aren't that good.)

**Comment by hzeller** [comment]
You can try connecting it [manually](https://github.com/hzeller/rpi-rgb-led-matrix/blob/master/wiring.md) to your pi and see if it works. There is no harm to try, it will just not work properly. Keep your data lines short though (~15cm) to not load the GPIO pins with too much impedance (the GPIO pins are directly connected to the internals of the processor).

**Comment by marcmerlin** [comment]
With apologies to everyone, this project is doing a one time cleanup of 
all the bugs that have not been addressed and closed in the last 10 years.

If you filed an issue to ask for help, please do not re-open this one or a new
bug, but go here instead: https://rpi-rgb-led-matrix.discourse.group/

Please be mindful that this project does not have any full time maintainers
or anyone paid. You should not expect personal help for your own projects or
answers to programming questions but you can ask for help with panels that
aren't working, setup questions, or anything that may be a bug here:
https://rpi-rgb-led-matrix.discourse.group/
If you need programming help and are willing to pay someone for it, please
state so in your message, and someone may be interested.

If you did report an actual bug that is still not fixed today, please accept
our apologies for this being mass closed, you can absolutely send a followup
and re-open the issue, or open a new up to date one as applicable.
Note that we very much welcome help, so if you can help fix your bug, you will
be a superstar :)

If your bug was linked to a pull request that was not resolved, please update
the pull request and the issue, and we will do our best to review them. We do
thank you for your undertanding that we have very limited free time, so any
extra work you do along with your bug report or PR request will be very much
appreciated and improve the odds that they will be looked at.

If you want to help contribute as a python or C# or any other language bindings,
you are also a superstar, please state so in the issue and we'll do our best to
give you the access and reviews you need.

For anything related to PWM panel support, please make sure you read README and
then refer to https://github.com/hzeller/rpi-rgb-led-matrix/issues/466 . 
Please do not comment if you're haven't read everything and/or if you're just
asking when your panel is going to be supported. Those bugs are if you're 
able to contribute to the effort.

For rPI5: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1603

For anything python, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
For anything C#, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
